sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -144,7 +141,7 @@ class ParquetToSparkSchemaConverter(
         originalType match {
           case INT_64 | null => LongType
           case DECIMAL => makeDecimalType(Decimal.MAX_LONG_DIGITS)
-          case UINT_64 => typeNotSupported()
+          case UINT_64 => DecimalType.LongDecimal
```

Comment:
I don't think it's related to `DecimalType.LongDecimal`. We want it to store uint64, so the precision we need is `java.lang.Long.toUnsignedString(-1).length`, which is 20.

Let's make it more explicit
```
// The precision to hold the largest unsigned long is: `java.lang.Long.toUnsignedString(-1).length` = 20
case UINT_64 => DecimalType(20, 0)
```

Reply:
OK, I will send a followup